pkg/controllers/resourceapply/resourceapply.go

```diff
@@ -224,3 +294,45 @@ func applyDaemonSet(ctx context.Context, client coreclientv1.Client, recorder re
 	recorder.Event(required, corev1.EventTypeNormal, "Updated successfully", "Resource was successfully updated")
 	return true, nil
 }
+
+func applyPodDisruptionBudget(ctx context.Context, client coreclientv1.Client, recorder record.EventRecorder, requiredOriginal *policyv1.PodDisruptionBudget) (bool, error) {
+	required := requiredOriginal.DeepCopy()
+
+	existing := &policyv1.PodDisruptionBudget{}
+	err := client.Get(ctx, coreclientv1.ObjectKeyFromObject(required), existing)
+	if apierrors.IsNotFound(err) {
+		err = client.Create(ctx, required)
+		if err != nil {
+			recorder.Event(required, corev1.EventTypeWarning, "Create failed", err.Error())
+			return false, err
+		}
+		recorder.Event(required, corev1.EventTypeNormal, "Created successfully", "Resource was successfully created")
+		return true, nil
+	}
+	if err != nil {
+		recorder.Event(required, corev1.EventTypeWarning, "Failed to get resource for update", err.Error())
+		return false, err
+	}
+
+	modified := resourcemerge.BoolPtr(false)
+	existingCopy := existing.DeepCopy()
+
+	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
+	contentSame := equality.Semantic.DeepEqual(existingCopy.Spec, required.Spec)
+
+	if !*modified && contentSame {
+		return false, nil
+	}
+
+	// at this point we know that we're going to perform a write.  We're just trying to get the object correct
+	toWrite := existingCopy // shallow copy so the code reads easier
+	toWrite.Spec = *required.Spec.DeepCopy()
+
+	err = client.Update(ctx, toWrite)
+	if err != nil {
```

Comment:
Nit, don't shadow previous error, declare within the if scope
```suggestion
	if err := client.Update(ctx, toWrite); err != nil {
```